Adafruit_Arcada_USBMSD.cpp

```diff
@@ -6,82 +6,85 @@ static uint32_t last_access_ms;
 
 #if defined(USE_TINYUSB)
 static Adafruit_USBD_MSC usb_msc;
+#endif
 
-#if defined(ARCADA_USE_QSPI_FS)
-  extern Adafruit_QSPI_Flash arcada_qspi_flash;
+extern FatFileSystem Arcada_QSPI_FileSys;
+extern Adafruit_SPIFlash Arcada_QSPI_Flash;
 
-  int32_t msc_write_cb (uint32_t lba, uint8_t* buffer, uint32_t bufsize);
-  int32_t msc_read_cb (uint32_t lba, void* buffer, uint32_t bufsize);
-  void msc_flush_cb (void);
-  void flash_cache_read (uint8_t* dst, uint32_t addr, uint32_t count);
-  uint32_t flash_cache_write (uint32_t dst, void const * src, uint32_t len);
-  void flash_cache_flush (void);
+int32_t qspi_msc_write_cb (uint32_t lba, uint8_t* buffer, uint32_t bufsize);
+int32_t qspi_msc_read_cb (uint32_t lba, void* buffer, uint32_t bufsize);
+void qspi_msc_flush_cb (void);
 
-  #define FLASH_CACHE_SIZE          4096        // must be a erasable page size
-  #define FLASH_CACHE_INVALID_ADDR  0xffffffff
+#if defined(ENABLE_EXTENDED_TRANSFER_CLASS)
+extern SdFatEX Arcada_SD_FileSys;
+#else
+extern SdFat Arcada_SD_FileSys;
+#endif
 
-  uint32_t cache_addr = FLASH_CACHE_INVALID_ADDR;
-  uint8_t  cache_buf[FLASH_CACHE_SIZE];
-#elif defined(ARCADA_USE_SD_FS)
-  extern SdFat FileSys;
+int32_t sd_msc_write_cb (uint32_t lba, uint8_t* buffer, uint32_t bufsize);
+int32_t sd_msc_read_cb (uint32_t lba, void* buffer, uint32_t bufsize);
+void sd_msc_flush_cb (void);
 
-  int32_t msc_write_cb (uint32_t lba, uint8_t* buffer, uint32_t bufsize);
-  int32_t msc_read_cb (uint32_t lba, void* buffer, uint32_t bufsize);
-  void msc_flush_cb (void);
-#endif // SD or QSPI
-#endif // TinyUSB
 
 /**************************************************************************/
 /*!
     @brief  Make the raw filesystem of the Arcada board available over USB
     @return True on success, false on failure
```

Comment:
add:
    @param desiredFilesys The filesystem we'd prefer to use, can be  ARCADA_FILESYS_SD, ARCADA_FILESYS_QSPI, or ARCADA_FILESYS_SD_AND_QSPI

Reply:
yep was asleep :)